Практична робота №3
Завдання 2
Авдєєва Серафима ФІТ 4-4

In [25]:
import pandas as pd

# 1: Зчитування CSV файлу
data = pd.read_csv("E:/University/виконані завдання/МН виконані/bestsellers with categories.csv", encoding='unicode_escape')

In [26]:
# 2: Виведення перших десяти рядків
print(data.head(10))

                                                Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   
5      A Dance with Dragons (A Song of Ice and Fire)   
6  A Game of Thrones / A Clash of Kings / A Storm...   
7                     A Gentleman in Moscow: A Novel   
8      A Higher Loyalty: Truth, Lies, and Leadership   
9                          A Man Called Ove: A Novel   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fic

In [27]:
# 3: Виведення розмірів датасету
print("Розмір датасету:", data.shape)

# скільки книг зберігає дані датасет?
book_count = data.shape[0]
print("Кількість книг у датасеті:", book_count)

unique_book_count = data["Name"].nunique()
print("Кількість книг у датасеті (без повторень):", unique_book_count)

Розмір датасету: (550, 7)
Кількість книг у датасеті: 550
Кількість книг у датасеті (без повторень): 351


In [28]:
# 4.Змінення назв усіх колонок: назва з маленької букви, без пробілів через підкреслення.
data.columns = ['name', 'author', 'user_rating', 'reviews', 'price', 'year', 'genre']

In [29]:
# 5. Визначення кількість пропусків у кожному стовпці
missing_values = data.isna().sum()
print("Кількість пропусків у кожному стовпці: ")
print(missing_values)

# Перевірка, чи є пропуски в змінних 
if missing_values.sum() > 0:
    print("Пропуски в змінних. є")
else:
    print("Пропусків у змінних немає.")


Кількість пропусків у кожному стовпці: 
name           0
author         0
user_rating    0
reviews        0
price          0
year           0
genre          0
dtype: int64
Пропусків у змінних немає.


In [30]:
# 6. Перевірка унікальних жанрів зі стовпця "genre"
unique_genres = data['genre'].unique()

# Виведення унікальних жанрів
print("Унікальні жанри:")
print(unique_genres)

Унікальні жанри:
['Non Fiction' 'Fiction']


In [31]:
# 7. Визначення максимальної, мінімальної, медіанної ціни
# за допомогою методів  max, min, mean, median

print("Максимальна ціна:", data['price'].max())
print("Мінімальна ціна:", data['price'].min())
print("Середня ціна:", data['price'].mean())
print("Медіанна ціна:", data['price'].median())


Максимальна ціна: 105
Мінімальна ціна: 0
Середня ціна: 13.1
Медіанна ціна: 11.0


In [32]:
# 8
# Який рейтинг у датасеті найвищий?
max_rating = data['user_rating'].max()
print("Найвищий рейтинг у датасеті:", max_rating)

# Скільки книг мають такий рейтинг?
books_max_rating = data[data['user_rating'] == max_rating]
count_books_max_rating = len(books_max_rating)
print("Кількість книг з найвищим рейтингом:", count_books_max_rating)

# Яка книга має найбільше відгуків?
book_max_reviews = data[data['reviews'] == data['reviews'].max()]
print("Книга з найбільшою кількістю відгуків:")
print(book_max_reviews[['name', 'reviews']])

# З тих книг, що потрапили до Топ-50 у 2010 році, яка книга найдорожча?
books_2010 = data[data['year'] == 2010]
most_expensive_2010 = books_2010[books_2010['price'] == books_2010['price'].max()]
print("Найдорожча книга серед книг 2010 року:")
print(most_expensive_2010[['name', 'price']])

# Скільки книг жанру Fiction потрапили до Топ-50 у 2012 році?
fiction_2012 = data[(data['year'] == 2012) & (data['genre'] == 'Fiction')]
count_fiction_2012 = len(fiction_2012)
print("Кількість книг жанру Fiction у 2012 році:", count_fiction_2012)

# Скільки книг з рейтингом 4.9 потрапило до рейтингу у 2010 та 2011 роках (використовуйте | або метод isin)?
rating_4_9 = data[(data['user_rating'] == 4.9) & (data['year'].isin([2010, 2011]))]
count_rating_4_9 = len(rating_4_9)
print("Кількість книг з рейтингом 4.9 у 2010 та 2011 роках:", count_rating_4_9)

Найвищий рейтинг у датасеті: 4.9
Кількість книг з найвищим рейтингом: 52
Книга з найбільшою кількістю відгуків:
                        name  reviews
534  Where the Crawdads Sing    87841
Найдорожча книга серед книг 2010 року:
                                                  name  price
272  Publication Manual of the American Psychologic...     46
Кількість книг жанру Fiction у 2012 році: 21
Кількість книг з рейтингом 4.9 у 2010 та 2011 роках: 1


In [33]:
# 9. Вивести максимальну і  мінімальну ціну для жанру Fiction I NonFiction 
# (за допомогою методів  groupby та agg, для підрахунку мінімальних та
# максимальних значень використовуйте max та min). 
# Використовуйте тількі потрібні вам стовпці

genre_prices = data.groupby('genre')['price'].agg(['max', 'min'])
print(genre_prices)

             max  min
genre                
Fiction       82    0
Non Fiction  105    0


Висновок

У ході виконання даної практичної роботи я вирішила завдання, пов'язані з аналізом даних у форматі csv. Було проведено роботу з даними про глобальну статистику YouTube за 2023 рік (файл Global YouTube Statistics.csv) та про найбільш продавані книги Amazon з 2009 по 2019 рік (файл bestsellers with categories.csv). Для цього було використано бібліотеки pandas, numpy й matplotlib. Я обробила дані, привівши їх, де це можливо, до типу float і замінивши пропуски на середні значення, побудувала діаграму розподілу переглядів та визначила, наприклад, кількість унікальних країн у першому файлі (49), країну з найбільшою кількістю відео на YouTube (США), максимальну (105), мінімальну(0) та медіанну(11) ціну та книгу з найбільшою кількістю відгуків (Where the Crawdads Sing) тощо.